tests/test_streams.rs

```diff
@@ -228,6 +228,16 @@ fn test_assorted_2() {
     let _: RedisResult<String> = con.xadd("k99", "1000-0", &[("a", "b"), ("c", "d")]);
     let _: RedisResult<String> = con.xadd("k99", "1000-1", &[("e", "f"), ("g", "h")]);
 
+    // test empty PEL
+    let empty_reply: StreamPendingReply = con.xpending("k99", "g99").unwrap();
+
+    assert_eq!(empty_reply.count(), 0);
+    if let StreamPendingReply::Empty = empty_reply {
```

Comment:
Did you try using  `assert_eq!(empty_reply, StreamPendingReply::Empty)` here? I'm guessing you'll need to add `derive(PartialEq)]` to `StreamPendingReply`? If this works, I would go with this approach and remove this if/else block. I'd rather we use assertions to fail vs. calling panic.

Reply:
Yes, this was something I encountered - if I derive PartialEq (which does indeed fix this) then the contained data structures would also need to derive PartialEq, including the Consumers vector etc. If you're happy with this, I'll change it to derive PartialEq.

Reply:
Makes sense. Let's skip adding the `derive PartialEq` until we absolutely need it.